spec/javascripts/reducers/pastIterations_spec.js

```diff
@@ -0,0 +1,128 @@
+import actionTypes from "actions/actionTypes";
+import reducer from "../../../app/assets/javascripts/reducers/pastIterations";
+
+describe("Past Iterations Reducer", () => {
+  describe("RECEIVE_PAST_ITERATIONS", () => {
+    it("returns formatted past iterations", () => {
+      const state = [];
+
+      const action = {
+        type: actionTypes.RECEIVE_PAST_ITERATIONS,
+        data: [{
+          hasStories: true,
+          iterationNumber: 1,
+          stories: []
+        }]
+      };
+
+      expect(reducer(state, action)).toEqual([{
+        hasStories: true,
+        iterationNumber: 1,
+        error: null,
+        storyIds: [],
+        fetched: false,
+        isFetching: false,
+        stories: undefined
+      }]);
+    });
+  });
+
+  describe("REQUEST_PAST_STORIES", () => {
+    it("adds isFetching: true, to the selected iteration ", () => {
+      const state = [
+        {
+          iterationNumber: 1
+        },
+        {
+          iterationNumber: 2
+        }
+      ];
+
+      const action = {
+        type: actionTypes.REQUEST_PAST_STORIES,
+        iterationNumber: 1,
+      };
+
+      expect(reducer(state, action)).toEqual([
+        {
+          iterationNumber: 1,
+          isFetching: true
+        },
+        {
+          iterationNumber: 2
+        }
+      ]);
+    });
+  });
+
+  describe("RECEIVE_PAST_STORIES", () => {
+    it("returns formatted past stories", () => {
+      const state = [
+        {
+          iterationNumber: 1
+        },
+        {
+          iterationNumber: 2
+        }
+      ];
+
+      const action = {
+        type: actionTypes.RECEIVE_PAST_STORIES,
+        iterationNumber: 1,
+        stories:[
+          {id: 1},
+          {id: 2}
+        ],
+        from: undefined
```

Comment:
Do we need this?

Reply:
No we don't, but I though make the action from the test more like the real action, would be a good idea. What do you think ?